base/src/account/mod.rs

```diff
@@ -85,4 +85,53 @@ impl mina_hasher::Hashable for Account {
     }
 }
 
-mina_merkle::impl_poseidon_legacy_hasher_pool_provider!(Account);
+/// TODO
+#[derive(Clone, Debug, AutoFrom)]
+#[auto_from(mina_serialization_types::account::AccountV0)]
+pub struct AccountHardFork {
+    /// Account public key
+    pub public_key: CompressedPubKey,
+    /// Account token ID
+    pub token_id: TokenId,
+    /// Permission associated with the given token
+    pub token_permissions: TokenPermissions,
+    /// Token Symbol
+    pub token_symbol: [u8; 32],
+    /// Balance of token held by account
+    pub balance: Amount,
+    /// Nonce (incremented with each tx to prevent replay)
+    pub nonce: AccountNonce,
+    /// ?
+    pub receipt_chain_hash: ChainHash,
+    /// Delegate for staking purposes
+    pub delegate: Option<CompressedPubKey>,
+    /// The state hash this account is voting for
+    pub voting_for: StateHash,
+    /// Any timing limitations places on this accounts balance
+    /// Used for vesting
+    pub timing: Timing,
+    /// Level of permission required to do different account actions
+    pub permissions: PermissionsHardFork,
+    /// TODO: This should contain a Snapp account data once we have something to test against
+    pub zkapp: Option<()>,
+    /// TODO: This should contain a Snapp account data once we have something to test against
+    pub zkuri: Option<()>,
+}
+
+impl mina_hasher::Hashable for AccountHardFork {
```

Comment:
Can we rename this to Account?, as we're not supporting backward compatible Account type

Reply:
Fixed